data/transactions/logic/crypto.go

```diff
@@ -70,6 +71,34 @@ func opSHA512_256(cx *EvalContext) error {
 	return nil
 }
 
+// Sumhash512 corresponds to the hash used in State Proofs
+func opSumhash512(cx *EvalContext) error {
+	last := len(cx.Stack) - 1
+	h := sumhash.New512(nil)
+	h.Write(cx.Stack[last].Bytes)
+	cx.Stack[last].Bytes = h.Sum(nil)
```

Comment:
by def
>	// Sum appends the current hash to b and returns the resulting slice.
	// It does not change the underlying hash state.
	Sum(b []byte) []byte

so why not appending into `cx.Stack[last].Bytes` ? in theory it could skip allocation?

Reply:
We have a rule that we never modify a stack value's slice in place, because they can be shared. (`dup`, for example, does not clone the slice, and I think several `txn FIELD` opcodes put a slice in there that perhaps puts the `txn` slice directly onto the stack, so modifying would modify the txn).